UserDetailsClient/UserDetailsClient.Core/MainPage.xaml.cs

```diff
@@ -16,27 +16,6 @@ public MainPage()
             InitializeComponent();
         }
 
-        protected override async void OnAppearing()
-        {
-            UpdateSignInState(false);
-
-            // Check to see if we have a User
-            // in the cache already.
-            try
-            {
-                AuthenticationResult ar = await App.PCA.AcquireTokenSilentAsync(App.Scopes, GetUserByPolicy(App.PCA.Users, App.PolicySignUpSignIn), App.Authority, false);
-                UpdateUserInfo(ar);
-                UpdateSignInState(true);
-            }
-            catch (Exception ex)
-            {
-                // Uncomment for debugging purposes
-                //await DisplayAlert($"Exception:", ex.ToString(), "Dismiss");
-
-                // Doesn't matter, we go in interactive mode
-                UpdateSignInState(false);
-            }
-        }
```

Comment:
When testing the embedded webview, the ATSilent was getting called numerous times (especially in iOS). 

Repo: 1. Start app (ATSilent is called) 2. Hit Sign-in and login page appears, login 3. ATSilent is called 4. App continues with AT interactive call. 

I removed the OnAppearing() method, and have just the interactive log-in. The App is much faster now because it's making less network calls. If we want to have the user logged in silently, we will need to do it another way.

Reply:
Consider if this code can be called as part of the initialization logic (right after IniitlizeComponent)

---
In reply to: [201172740](https://github.com/Azure-Samples/active-directory-b2c-xamarin-native/pull/59#discussion_r201172740) [](ancestors = 201172740)